core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -141,7 +141,9 @@ class GroupMetadataManager(val brokerId: Int,
       // if we crash or leaders move) since the new leaders will still expire the consumers with heartbeat and
       // retry removing this group.
       val groupPartition = partitionFor(group.groupId)
-      val tombstone = new Message(bytes = null, key = GroupMetadataManager.groupMetadataKey(group.groupId))
+      // Tombstone does not care about message magic value, simply use magic value 0.
+      val tombstone = new Message(bytes = null, key = GroupMetadataManager.groupMetadataKey(group.groupId),
+        timestamp = Message.NoTimestamp, magicValue = Message.MagicValue_V0)
```

Comment:
That's not what I meant. Since this message will be eventually appended to the log through Log.append, if it's compressed and of format v0, we need to recompress it. If it's format v1, the recompression can be avoid. Also, since v1 carries more metadata, it seems that we should always try to use message v1 if possible.